src/main/java/uk/gov/digital/ho/hocs/domain/service/MessageLogService.java

```diff
@@ -44,6 +44,11 @@ public void updateStatus(String messageId, Status status) {
         messageLogRepository.updateStatus(messageId, status);
     }
 
+    @Transactional
+    public void completeWithStatus(String messageId, Status status) {
```

Comment:
This makes sense, does this update the `completed` column with the timestamp?

Reply:
It does - resolved.

Reply:
Yes.

Though in writing up my answer to your deleted comment I think it might be cleaner to have:
* MigrationService.createMigrationCase set the status (completed, duplicate, etc.)
* MigrationMessageHandler just set the completed timestamp and not the status, so this wouldn't be needed

Thoughts?

Reply:
Its a bit of a weird one, the final status and the completed timestamp imo should be done at the same time, in my head it reads "this is the final state change and it happened here". Although it depends on if we see completed as the case action or the message completion.